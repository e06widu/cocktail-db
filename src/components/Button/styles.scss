.custom-button {
    padding: 5px 12px;
    border-radius: 3px;
    color: #fff;
    line-height: 24px;
    display: inline-block;
    background-color: #c52d2f;
    cursor: pointer;
    border: none;
    position: relative;
    overflow: hidden;
    transition: background-color 0.3s, transform 0.3s;
}

.custom-button::after {
    content: '';
    position: absolute;
    bottom: 0;
    left: 0;
    width: 0%;
    height: 3px;
    background-color: #fff;
    transition: width 0.3s;
}

.custom-button:hover {
    background-color: #ff5b5e;
}

.custom-button:hover::after {
    width: 100%;
}

.custom-button:focus {
    outline: none;
}

.custom-button:active {
    transform: scale(0.95);
    /* Add scale transform */
}

.custom-button:active::after {
    width: 100%;
}