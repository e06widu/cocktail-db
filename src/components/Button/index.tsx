import React, { ReactNode } from 'react';
import './styles.scss';

interface ButtonProps {
  onClick: () => void;
  children: ReactNode;
  icon?: ReactNode;
  disabled?: boolean;
}

const Button: React.FC<ButtonProps> = ({ onClick, children, icon, disabled }) => {
  return (
    <button className={`custom-button ${disabled ? 'disabled' : ''}`} onClick={onClick} disabled={disabled}>
      {icon && <span className="button-icon">{icon}</span>}
      {children}
    </button>
  );
};

export default Button;