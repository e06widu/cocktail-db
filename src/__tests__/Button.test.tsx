import { fireEvent, render } from '@testing-library/react';
import Button from '../components/Button';
import '@testing-library/jest-dom/extend-expect';

jest.mock('../components/styles.scss', () => ({}));

describe('Button', () => {
    it('renders the button correctly', () => {
      const onClickMock = jest.fn();
      const { getByText } = render(
        <Button onClick={onClickMock}>Click me</Button>
      );
      const buttonElement = getByText('Click me');
  
      expect(buttonElement).toBeInTheDocument();
      expect(buttonElement.tagName).toBe('BUTTON');
      expect(buttonElement).not.toBeDisabled();
  
      // Snapshot testing
      expect(buttonElement).toMatchSnapshot();
    });
  
    it('calls the onClick function when clicked', () => {
      const onClickMock = jest.fn();
      const { getByText } = render(
        <Button onClick={onClickMock}>Click me</Button>
      );
      const buttonElement = getByText('Click me');
  
      fireEvent.click(buttonElement);
      expect(onClickMock).toHaveBeenCalledTimes(1);
  
      // Snapshot testing
      expect(buttonElement).toMatchSnapshot();
    });
  
    it('renders the icon when provided', () => {
      const onClickMock = jest.fn();
      const { getByTestId } = render(
        <Button onClick={onClickMock} icon={<span data-testid="test-icon">Icon</span>}>Click me</Button>
      );
      const iconElement = getByTestId('test-icon');
  
      expect(iconElement).toBeInTheDocument();
  
      // Snapshot testing
      expect(iconElement).toMatchSnapshot();
    });
  
    it('disables the button when disabled prop is true', () => {
      const onClickMock = jest.fn();
      const { getByText } = render(
        <Button onClick={onClickMock} disabled={true}>Click me</Button>
      );
      const buttonElement = getByText('Click me');
  
      expect(buttonElement).toBeDisabled();
  
      // Snapshot testing
      expect(buttonElement).toMatchSnapshot();
    });
  });