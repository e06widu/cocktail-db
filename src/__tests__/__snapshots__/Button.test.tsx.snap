// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button calls the onClick function when clicked 1`] = `
<button
  class="custom-button "
>
  Click me
</button>
`;

exports[`Button disables the button when disabled prop is true 1`] = `
<button
  class="custom-button disabled"
  disabled=""
>
  Click me
</button>
`;

exports[`Button renders the button correctly 1`] = `
<button
  class="custom-button "
>
  Click me
</button>
`;

exports[`Button renders the icon when provided 1`] = `
<span
  data-testid="test-icon"
>
  Icon
</span>
`;
